### DISCUSSION TITLE
Open Bug Fixing

### DISCUSSION THREAD

**Discussion by sparsick** [support-and-care/maven-support-and-care#23] [Ideas for Funding]
Looking at this [filter](https://issues.apache.org/jira/issues/?filter=-5&jql=project%20in%20(MNG%2C%20SUREFIRE%2C%20MASFRES%2C%20MTOMCAT%2C%20MANTRUN%2C%20ARCHETYPE%2C%20MARCHETYPES%2C%20MARTIFACT%2C%20MASSEMBLY%2C%20MBUILDCACHE%2C%20MCHANGELOG%2C%20MCHANGES%2C%20MCHECKSTYLE%2C%20MCLEAN%2C%20MCOMPILER%2C%20MDEP%2C%20MDEPLOY%2C%20MDOAP%2C%20MDOCCK%2C%20DOXIA%2C%20DOXIASITETOOLS%2C%20DOXIATOOLS%2C%20MEAR%2C%20MEJB%2C%20MENFORCER%2C%20MGPG%2C%20MPH%2C%20MINDEXER%2C%20MINSTALL%2C%20MINVOKER%2C%20MJAR%2C%20MJARSIGNER%2C%20MJAVADOC%2C%20MJDEPRSCAN%2C%20MJDEPS%2C%20MJLINK%2C%20MJMOD%2C%20JXR%2C%20MLINKCHECK%2C%20MPATCH%2C%20MPDF%2C%20MPLUGINTESTING%2C%20MPLUGIN%2C%20MPMD%2C%20MPOM%2C%20MPIR%2C%20MNGSITE%2C%20MRAR%2C%20MRELEASE%2C%20MRRESOURCES%2C%20MREPOSITORY%2C%20MRESOLVER%2C%20MRESOURCES%2C%20SCM%2C%20MSCMPUB%2C%20MSCRIPTING%2C%20MSHADE%2C%20MSHARED%2C%20MSITE%2C%20MSKINS%2C%20MSOURCES%2C%20MSTAGE%2C%20MTOOLCHAINS%2C%20MVERIFIER%2C%20WAGON%2C%20MWAR%2C%20MWRAPPER%2C%20MMETRIC)%20AND%20issuetype%20%3D%20Bug%20AND%20resolution%20%3D%20Unresolved%20order%20by%20priority%20DESC%2Cupdated%20DESC) that search for all open bug, there exists a huge number of open bugs.

Some ideas for what to do with them:
- Doing a backlog grooming and deciding if this bug can be closed with "Won't fix" and a reason why
- Fix them (priority can be done by votes, some bugs have already votes)
-

**Comment by slawekjaranowski** [comment]
There is a [filter / search](https://issues.apache.org/jira/issues/?jql=category%20%3D%20Maven%20and%20resolution%20%3D%20Unresolved) for all open issues from category Maven

**Comment by taijidude** [comment]
And maybe some bugs could be marked with some tags like "beginner" or "good First issue".

**Comment by hendrikebbers** [comment]
@slawekjaranowski @olamy @rmannibucau @taijidude who would be the best person to create some epics and group issues. Too many for me to have an overview in a reasonable time but maybe someone of you can create some groups? By doing so we could add a discussion here per group to have it on our list.

**Reply by slawekjaranowski** [reply]
In ASF JIRA I can not see a epic features.

We have a many JIRA projects:
  https://issues.apache.org/jira/secure/BrowseProjects.jspa?selectedCategory=10510&selectedProjectType=all&sortColumn=name&sortOrder=ascending

We can work project by project as group .... in my opinion it will be the most effective

We can use a fix version field, like:
  https://issues.apache.org/jira/issues/?jql=category%20%3D%20Maven%20and%20resolution%20%3D%20Unresolved%20and%20fixVersion%20%3D%20waiting-for-feedback

or labels like in:
  https://issues.apache.org/jira/issues/?jql=category%20%3D%20Maven%20and%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20up-for-grabs

By the way first we should think of what we want to result of such task.
I now it is and will be never ending story.

**Comment by olamy** [comment]
sounds possible to add Epic. We only need to ask ASF infra to setup this for our projects.
if we really need this, I can ask. But for all projects?

**Reply by hendrikebbers** [reply]
I assume it would make sense to have epics at https://issues.apache.org/jira/projects/MNG/issues/MNG-8009?filter=allopenissues If an issue is like "Check XYZ for all prominent plugins" I would say it is fine to have it in MNG space and than link the other issues (in plugins) to it.

**Reply by olamy** [reply]
I created an Apache Infra task to add capability to create Epic (https://issues.apache.org/jira/browse/INFRA-25822)

**Comment by slawekjaranowski** [comment]
Can we put here more information how we want to use epic for open bugs, how it can help?

**Reply by hendrikebbers** [reply]
Let me add my thoughts on that topic:

Support & Care will hopefully get funding for N topics (based on the discussions in this repo). Once we know what topics will be funded, we will create more concrete issues for those topics. Let's assume we split a topic into 10 issues since it affects multiple plugins, etc. We will create all those issues directly in the Apache Jira. However, next to that, we need documentation of the overall state of the topic (including all issues). This is at least needed to get an overview of the funded topics and to communicate the state to the German government to get paid for finished topics. The idea was to create an epic in Jira and then link all the technical issues. If that will end in paperwork, a lot of mails, whatever, for me, it will be fine to use this repository to create issues that act as epics and just link all the Jira issues in the GitHub epic issues. I just need a place to keep track of the state and progress of the epics to communicate that.

**Reply by slawekjaranowski** [reply]
Ok it is more clear now what a epics will be for.

I focused on discussion subject `Open Bug Fixing` and thinking how we want to work on it.

Maybe we need another discussion - how funding will be reported.

**Comment by ascheman** [comment]
This discussion has been converted to Epic(s) for tracking:
- #42
- #43

Closing this discussion.
